util/ranger/points.go

```diff
@@ -265,6 +281,59 @@ func (r *builder) buildFormBinOp(expr *expression.ScalarFunction) []point {
 	return nil
 }
 
+// HandlePadCharToFullLength handles the "PAD_CHAR_TO_FULL_LENGTH" sql mode for
+// CHAR[N] index columns.
+// NOTE: kv.ErrNotExist is returned to indicate that this value can not match
+//		 any (key, value) pair in tikv storage. This error should be handled by
+//		 the caller.
+func HandlePadCharToFullLength(sc *stmtctx.StatementContext, ft *types.FieldType, val types.Datum) (types.Datum, error) {
+	targetStr, err := val.ToString()
```

Comment:
Can we only call `TopString` for the needed types?

Reply:
Comment addressed, PTAL. Thanks.